sgt-back-end/server.ts

```diff
@@ -0,0 +1,191 @@
+import pg from 'pg';
+import express from 'express';
+import { ClientError } from './client-error.js';
+import { errorMiddleware } from './error-middleware.js';
+
+// only create ONE pool for your whole server
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+app.use(express.json());
+
+app.get('/api/grades/:gradeId', async (req, res, next) => {
+  try {
+    const gradeId = Number(req.params.gradeId);
+    if (!Number.isInteger(gradeId) || gradeId <= 0) {
+      throw new ClientError(400, '"gradeId" must be a positive integer');
+    }
+
+    const sql = `
+      select *
+        from "grades"
+        where "gradeId" = $1
+    `;
+    const params = [gradeId];
+    const result = await db.query(sql, params);
+    const grade = result.rows[0];
+    if (!grade) {
+      throw new ClientError(404, `Cannot find grade with "gradeId" ${gradeId}`);
+    }
+    res.json(grade);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/grades', async (req, res, next) => {
+  try {
+    const sql = `
+      select *
+        from "grades"
+    `;
+    const result = await db.query(sql);
+    const grades = Object.values(result.rows);
+    if (!grades) {
+      throw new ClientError(
+        500,
+        `An unexpected error has occured with our database.`
```

Comment:
Usually you'd only use template strings (`) only if you need them, which you don't here.